core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/session/GitConflictSessionListener.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2021 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.platform.sirius.ui.session;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.business.api.session.SessionListener;
+import org.eclipse.sirius.business.api.session.SessionManagerListener;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+
+public class GitConflictSessionListener extends SessionManagerListener.Stub {
+  @Override
+  public void notify(Session updatedSession, int notification) {
+    if (notification == SessionListener.ABOUT_TO_BE_REPLACED) {
```

Comment:
Yes, customizing ReloadingPolicy is a good idea. It can be used to avoid the reloading of resource which leads to some errors (since the resource is in conflict state).

Reply:
I'll also add a JUnit with a model under a Git repository. We'll see how the test framework's gonna handle this :).

Reply:
Finally adding a Git submodule (especially in a conflict state) is not simple. The test is removed for now.